quinn-proto/src/shared.rs

```diff
@@ -257,33 +273,81 @@ pub struct ServerConfig<S>
 {
-    /// Transport configuration to use for incoming connections
+    /// TLS configuration used for incoming connections.
     pub transport: Arc<TransportConfig>,
+    /// Mutable TLS configuration used for incoming connections.
+    pub crypto: S::ServerConfig,
+    /// Private key used to send authenticated connection resets to peers who were
+    // communicating with a previous instance of this endpoint.
+    pub(crate) token_key: Vec<u8>,
+    // Whether clients have to prove ownership of an address before committing resources.
+    pub(crate) use_stateless_retry: bool,
+    /// The microseconds after a stateless retry token is going to be issued to consider it valid.
+    pub(crate) retry_token_lifetime: u64,
+    /// The maximal supported incoming connection buffer length.
+    pub(crate) accept_buffer: u32,
+    /// Whether clients are allowed to migrate to new addresses.
+    pub(crate) migration: bool,
+}
 
-    /// TLS configuration used for incoming connections.
+impl<S> ServerConfig<S>
+where
+    S: crypto::Session,
+{
+    /// Set transport configuration to use for incoming connections.
+    pub fn with_transport_config(&mut self, transport_config: TransportConfig) -> &mut Self {
+        self.transport = Arc::new(transport_config);
+        self
+    }
+
+    /// Set the TLS configuration used for incoming connections.
     ///
     /// Must be set to use TLS 1.3 only.
-    pub crypto: S::ServerConfig,
+    pub fn with_crypto(&mut self, crypto: S::ServerConfig) -> &mut Self {
+        self.crypto = crypto;
+        self
+    }
 
-    /// Private key used to authenticate data included in handshake tokens.
-    pub token_key: Vec<u8>,
-    /// Whether to require clients to prove ownership of an address before committing resources.
+    /// Set the private key used to authenticate data included in handshake tokens.
+    pub fn with_token_key(&mut self, token_key: Vec<u8>) -> &mut Self {
+        self.token_key = token_key;
+        self
+    }
+
+    /// Set whether to require clients to prove ownership of an address before committing resources.
     ///
     /// Introduces an additional round-trip to the handshake to make denial of service attacks more difficult.
-    pub use_stateless_retry: bool,
-    /// Microseconds after a stateless retry token was issued for which it's considered valid.
-    pub retry_token_lifetime: u64,
+    ///
+    /// This value is disabled by default.
+    pub fn with_stateless_retry(&mut self, enabled: bool) -> &mut Self {
+        self.use_stateless_retry = enabled;
+        self
+    }
+
+    /// Set microseconds after a stateless retry token was issued for which it's considered valid.
+    pub fn with_retry_token_lifetime(&mut self, retry_token_lifetime: u64) -> &mut Self {
+        self.retry_token_lifetime = retry_token_lifetime;
+        self
+    }
 
-    /// Maximum number of incoming connections to buffer.
+    /// Set the maximum supported length for the incoming connection buffer.
```

Comment:
I would like to stick to the same naming convention. Which for a set method is 'singular verb' + what the method does.  This is the convention used in most places. Do you have a proposal?

Reply:
I'm not sure I understand. This comment is discussing the wording of the comment on the line it's attached to.

Reply:
I found: "of incoming connections to buffer." a bit unclear, because 1) to what buffer, do I need to know there is a buffer and does it make sense for me to know. 2) The sentence just correct English, what do you mean with "connections to buffer"

Reply:
In the original wording, "buffer" is a verb, meaning roughly "to collect for future processing." While some discussion of what that means in this specific case may be justified, I don't think the changed wording adds any additional information.